bpf/lib/lb.h

```diff
@@ -674,10 +687,20 @@ static __always_inline int lb6_xlate(struct __ctx_buff *ctx,
 
 l4_xlate:
 	if (likely(backend->port) && key->dport != backend->port &&
-	    (nexthdr == IPPROTO_TCP || nexthdr == IPPROTO_UDP)) {
+			(nexthdr == IPPROTO_TCP ||
+#ifdef ENABLE_SCTP
+			 nexthdr == IPPROTO_SCTP ||
+#endif  /* ENABLE_SCTP */
+			 nexthdr == IPPROTO_UDP)) {
 		__be16 tmp = backend->port;
 		int ret;
-
+#ifdef ENABLE_SCTP
+		/* This will change the SCTP checksum, which we cannot fix right now.
+		 * This will likely need kernel changes before we can drop this.
+		 */
+		if (nexthdr == IPPROTO_SCTP)
+			return DROP_CSUM_L4;
+#endif  /* ENABLE_SCTP */
```

Comment:
We could check for this right after `l4_xlate:` to avoid having to modify the existing condition.

Reply:
Yep, it doesn't sound like port translation is right around the corner (given it requires kernel changes), so best to improve now I think.

Reply:
Ack. Done.